File: codecheck-case/lua/BKCT_IdenticalBranches/BKCT_IdenticalBranches_01.lua
function bad_01(x, y, z)
    -- bad
    if x == 2 then
        y = 32
        z = y * 2
    else
        y = 32
        z = y * 2
    end
end

function bad_02(condition)
    -- bad
    if condition then
        print("Repeated code")
    else
        print("Repeated code")
    end
end

function good_01(condition)
    -- good
    if condition then
        print("Repeated code")
    else
        print("Another different branch")
    end
end

File: codecheck-case/java/BKCT_IdenticalBranches/BKCT_IdenticalBranches_01.java
package java.BKCT_IdenticalBranches;

public class BKCT_IdenticalBranches_01 {
    void bad_01(int x, int y, int z) {
        // bad
        if (x == 2) {
            y = 32;
            z = y * 2;
        } else {
            y = 32;
            z = y * 2;
        }
    }

    void bad_02(boolean condition) {
        // bad
        if (condition) {
            System.out.println("Repeated code");
        } else {
            System.out.println("Repeated code");
        }
    }

    void good_01(boolean condition) {
        // good
        if (condition) {
            System.out.println("Repeated code");
        } else {
            System.out.println("Another different branch");
        }
    }
}